elip-deterministic-blinding-key.mediawiki

```diff
@@ -0,0 +1,122 @@
+which usually consist of multiple xpubs,
+and thus multiple BIP32 seeds from which a descriptor blinding key could be derived.
+SLIP77 also does not work well when there are multiple BIP44 accounts,
+as they will have the same master blinding key.
+
+This document defines a standard way to deterministically derive a descriptor blinding key from an ordinary descriptor.
+
+Deriving the descriptor blinding key from the ordinary descriptor means that multisig participants can avoid needing a key construction and sharing protocol.
+
+==Design==
+
+===Overview===
+We propose a new possible value for the descriptor blinding key
+
+<pre>
+ct(elipXXXX,<DESCRIPTOR>)
+</pre>
+
+Which indicates that descriptor blinding key to be used is a '''view key''' deterministically derived from the ordinary descriptor <code>DESCRIPTOR</code>
+
+===Drawbacks===
+If a wallet uses this standard to derive its descriptor blinding key, anyone knowing the ordinary descriptor will be able to unblind all the corresponding outputs.
+
+Potential adopters of this standard should consider whether this behavior is acceptable.
+
+===Specification===
+The following descriptor:
+
+<pre>
+ct(elipXXXX,<DESCRIPTOR>)
+</pre>
+
+Is equivalent to:
+
+<pre>
+ct(<KEY>,<DESCRIPTOR>)
+</pre>
+
+Where <code>KEY</code> is a view blinding key derived as follows:
+* If the top-level script expression is <code>combo</code>, fail.
+* Future extensions that support multiple descriptors other than multi-path are not supported.
+* If <code>DESCRIPTOR</code> is multi-path, expand to every single descriptor.
+* For each single descriptor:
+** If it has wildcards, get the definite descriptor at index 2<sup>31</sup>-1<ref>
+We want to use a <code>scriptPubKey</code> that will not be used in a transaction.
+Otherwise it can be used directly by anyone to derive the descriptor blinding key.
+Wallets use <code>scriptPubKey</code>s starting from index 0,
+thus we choose the last non-hardened index,
+which in practice will never be used in a transaction.
+</ref>
+** Derive the <code>scriptPubKey</code> from the definite descriptor<ref>
+Here we use the <code>scriptPubKey</code> because descriptors have fields that might change without affecting the spending conditions.
+For instance changing the fingerprint of one xpub yields the same <code>scriptPubKey</code>s.
+Using the <code>scriptPubKey</code> allows us to treat such descriptors as equivalent.
+</ref>
+* Concatenate all <code>scriptPubKey</code>s obtained before.
+* Let ''H<sub>tag</sub>'' be a [https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki#Design BIP-340 tagged hash] with tag <code>Deterministic-View-Key/1.0</code><ref>
+A tagged hash is used to make sure hashes used in this context can't be reinterpreted in another one.
+</ref>
+* The descriptor blinding key is the tagged hash of the concatenated <code>scriptPubKey</code>s, ''<code>KEY</code> = H<sub>tag</sub>(<code>scriptPubKey</code>, ...) mod n'', converted to WIF format.
+
```

Comment:
I think you should specify here that the resulting key is a `bare key` as described in ELIP 0150, and must be tweaked according to those rules before being used for a given address. In particular, the key is not to be used as a SLIP77 style key.

The test vectors should give a solution for an arbitrary wildcard index with the resulting tweaked blinding key and address generated so implementations can verify their behaviour in this regard.

Reply:
The key is a private `bare` key, which is called `view` key.

Changed the line above to:
```diff
- Where <code>KEY</code> is a view blinding key derived as follows:
+ Where <code>KEY</code> is a ELIP-150 view blinding key derived as follows:
```

I'll add an address in the test vector later (although some test vectors do not have wildcards)

Reply:
Thanks. Even without wildcards, the final derived blinding key for the address will be tweaked and so IMO should be included in the test vector data.

Reply:
Added first address in the test vectors